lib/Dialect/Torch/Transforms/DecomposeComplexOps.cpp

```diff
@@ -1096,19 +1143,23 @@ class DecomposeAtenNativeBatchNormOp
     // 2. bias = bias.view(1, C, 1?, 1?, 1?)
     // 3. output = normalizedInput * weight + bias
     Value batchNormOutput = normalizedInput;
-    if (!weight.getType().isa<Torch::NoneType>()) {
-      // Rank of `weight` must be exactly 1.
+    if (!failed(checkNotNone(rewriter, op, weight))) {
+      // The shape of the `weight` tensor must be (numFeatures).
       if (getTensorRank(weight) != 1)
         return rewriter.notifyMatchFailure(op, "expected weight to be rank 1");
+      dim0EqualsNumFeatures(weight);
+
       weight = rewriter.create<AtenViewOp>(loc, reshapeType, weight,
                                            runningStatsSizeList);
       batchNormOutput = rewriter.create<AtenMulTensorOp>(
           loc, batchNormOutput.getType(), batchNormOutput, weight);
     }
-    if (!bias.getType().isa<Torch::NoneType>()) {
-      // Rank of `bias` must be exactly 1.
+    if (!failed(checkNotNone(rewriter, op, bias))) {
```

Comment:
I think just checking `bias.getType().isa<BaseTensorType>()` would be more straight forward.

Reply:
We would need another guard check `bias.getType().isa<OptionalType>()` and failure return because OptionalType can't be handled at all while with the current code, OptionalType is handled the same way as NoneType.

Reply:
Anyway I have kept the `checkNotNone` as a utility.